tests/Unit/TestCreation.hpp

```diff
@@ -12,43 +13,110 @@
-  return options.template get<TestCreation_detail::Opt<T>, Metavariables>();
+  Options<tmpl::list<OptionTag>> options("");
+  options.parse(
+      TestCreation_detail::AddGroups<OptionTag>::apply(construction_string));
+  return options.template get<OptionTag, Metavariables>();
 }
 
 /// \ingroup TestingFrameworkGroup
-/// Construct a factory object from a given string.  Each line in the
-/// string must be indented.
-template <typename BaseClass, typename Metavariables = NoSuchType>
+/// Construct a factory object from a given string.
+///
+/// The string must contain the name of the option (specifically, the struct
+/// being created from options, which is the name of the struct by default if
+/// no `name()` function is present). Note that it is not the name of the base
+/// class, but the name of the derived class that must be given. For example, to
+/// create a `BaseClass*` pointing to a derived class of type
+/// `size_t_argument_base` with an option tag name `SizeT` the following would
+/// be used:
+///
+/// \snippet Test_TestCreation.cpp size_t_argument_base
+///
+/// Option tags can be tested by passing them as the second template parameter.
+/// If the metavariables are required to create the class then the metavariables
+/// must be passed as the third template parameter. The default option tag is
+/// `TestCreationOpt<std::unique_ptr<BaseClass>>`.
+template <typename BaseClass,
+          typename OptionTag = TestCreationOpt<std::unique_ptr<BaseClass>>,
+          typename Metavariables = NoSuchType>
 std::unique_ptr<BaseClass> test_factory_creation(
     const std::string& construction_string) noexcept {
-  return TestHelpers::test_creation<std::unique_ptr<BaseClass>, Metavariables>(
-      construction_string);
+  return TestHelpers::test_creation<std::unique_ptr<BaseClass>, OptionTag,
+                                    Metavariables>(construction_string);
 }
 
 /// \ingroup TestingFrameworkGroup
 /// Construct an enum from a given string.
 ///
+/// The string must not contain the name of the enum being constructed. The
+/// following is an example of an enum named `Color` with a member `Purple`
+/// being created from options.
+///
+/// \snippet Test_TestCreation.cpp enum_purple
+///
 /// Whereas `test_creation` creates a class with options, this creates an enum.
 /// The enum is created from a simple string with no newlines or indents.
-template <typename T, typename Metavariables = NoSuchType,
+///
+/// Option tags can be tested by passing them as the second template parameter.
+/// If the metavariables are required to create the class then the metavariables
+/// must be passed as the third template parameter. The default option tag is
+/// `TestCreationOpt<T>`.
+template <typename T, typename OptionTag = TestCreationOpt<T>,
+          typename Metavariables = NoSuchType,
           Requires<std::is_enum<T>::value> = nullptr>
 T test_enum_creation(const std::string& enum_string) noexcept {
```

Comment:
This function is now identical to `test_creation`.  (And its documentation claims that it is not.)

Reply:
I'm really not sure what you want me to change. Could you please clarify?

Reply:
Probably just remove the function, as it is identical to the more common testing function above (except restricted to only work on enums).  If you would prefer not to do that, change the documentation to correctly describe the implemented behavior.

Reply:
Thanks for the clarification!